### ISSUE TITLE
clairctl analyse finds vulnerabilities but generated reports is show only one layer vulnerabilities

### ISSUE THREAD

**Issue by lcgkm** [jgsqware/clairctl#27]
```
% clairctl analyze hyper.cd/occ/b2c-mono-db: 3a80cacc7bbd894819a562d549ebc8ea7b345342
Image: hyper.cd/hyper.cd/occ/b2c-mono-db:3a80cacc7bbd894819a562d549ebc8ea7b345342
 4 layers found
 
  ➜ Analysis [sha256:038fb] found 113 vulnerabilities.
  ➜ Analysis [sha256:9278e] found 113 vulnerabilities.
  ➜ Analysis [sha256:d5194] found 113 vulnerabilities.
  ➜ Analysis [sha256:357ea] found 48 vulnerabilities.
```

But we can find only one layer vulnerabilities from report
```
% clairctl report hyper.cd/occ/b2c-mono-db: 3a80cacc7bbd894819a562d549ebc8ea7b345342
HTML report at ./reports/html/analysis-hyper.cd-occ-b2c-mono-db-3a80cacc7bbd894819a562d549ebc8ea7b345342.html
```

Report Content:
<img width="1280" alt="screen shot 2017-04-12 at 11 41 16 am" src="https://cloud.githubusercontent.com/assets/5523451/24940338/16c5ace0-1f75-11e7-8bc4-ba118a000398.png">

**Comment by davidxia** [comment]
I have the same question. Is there a reason `clairctl report` only generates a report for the last layer?

**Comment by davidxia** [comment]
Sorry, ignore my comment. According to [clair API docs](https://github.com/coreos/clair/blob/master/Documentation/api_v1.md#get-layersname):

> The GET route for the Layers resource displays a Layer and optionally all of its features and vulnerabilities. For an image composed of three layers A->B->C, calling this route on the third layer (C) will returns all the features and vulnerabilities for the entire image, including the analysis data gathered from the parent layers (A, B). For instance, a feature (and its potential vulnerabilities) detected in the first layer (A) will be shown when querying the third layer (C). On the other hand, a feature detected in the first layer (A) but then removed in either following layers (B, C) will not appear.

So we only need to analyze the most recent layer. Clairctl is correct in only generating a report for one layer, but it analyzes the wrong layer. It analyzes the oldest one instead of the most recent one.

I have a fix here. https://github.com/jgsqware/clairctl/pull/33

**Comment by jgsqware** [comment]
fixed in #33
